nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -182,40 +187,51 @@ Nav2Panel::Nav2Panel(QWidget * parent)
   runningTransition->setTargetState(idle_);
   running_->addTransition(runningTransition);
 
-  initial_thread_ = new InitialThread(client_nav_, client_loc_);
-  connect(initial_thread_, &InitialThread::finished, initial_thread_, &QObject::deleteLater);
+  nav_online_thread_ = new CheckOnlineThread(client_nav_);
+  loc_online_thread_ = new CheckOnlineThread(client_loc_);
+  nav_online_timer_ = new QTimer();
+  loc_online_timer_ = new QTimer();
+
+  QObject::connect(nav_online_timer_, SIGNAL(timeout()), nav_online_thread_, SLOT(start()));
+  QObject::connect(nav_online_thread_, SIGNAL(isActive()), client_nav_state_indicator_, SLOT(setActive()));
+  QObject::connect(nav_online_thread_, SIGNAL(isInactive()), client_nav_state_indicator_, SLOT(setInactive()));
+  QObject::connect(nav_online_thread_, SIGNAL(timedOut()), client_nav_state_indicator_, SLOT(setTimedOut()));
+  QObject::connect(paused_, SIGNAL(entered()), client_nav_state_indicator_, SLOT(setInactive()));
+  QObject::connect(paused_, SIGNAL(exited()), client_nav_state_indicator_, SLOT(setActive()));
+
+  QObject::connect(loc_online_timer_, SIGNAL(timeout()), loc_online_thread_, SLOT(start()));
+  QObject::connect(loc_online_thread_, SIGNAL(isActive()), client_loc_state_indicator_, SLOT(setActive()));
+  QObject::connect(loc_online_thread_, SIGNAL(isInactive()), client_loc_state_indicator_, SLOT(setInactive()));
+  QObject::connect(loc_online_thread_, SIGNAL(timedOut()), client_loc_state_indicator_, SLOT(setTimedOut()));
+  QObject::connect(paused_, SIGNAL(entered()), client_loc_state_indicator_, SLOT(setInactive()));
+  QObject::connect(paused_, SIGNAL(exited()), client_loc_state_indicator_, SLOT(setActive()));
 
   QSignalTransition * activeSignal = new QSignalTransition(
-    initial_thread_,
-    &InitialThread::activeSystem);
-  activeSignal->setTargetState(idle_);
-  pre_initial_->addTransition(activeSignal);
+    nav_online_thread_,
+    &CheckOnlineThread::isActive);
+  activeSignal->setTargetState(online_);
+  offline_->addTransition(activeSignal);
 
   QSignalTransition * inactiveSignal = new QSignalTransition(
-    initial_thread_,
-    &InitialThread::inactiveSystem);
-  inactiveSignal->setTargetState(initial_);
-  pre_initial_->addTransition(inactiveSignal);
-
-  state_machine_.addState(pre_initial_);
-  state_machine_.addState(initial_);
-  state_machine_.addState(idle_);
-  state_machine_.addState(running_);
-  state_machine_.addState(canceled_);
-  state_machine_.addState(reset_);
+    nav_online_thread_,
+    &CheckOnlineThread::isInactive);
+  inactiveSignal->setTargetState(offline_);
+  online_->addTransition(inactiveSignal);
+
+  state_machine_.addState(offline_);
```

Comment:
It looks like you removed alot of the states - these are necessary for navigation. Can you explain what you are trying to accomplish?

Reply:
If possible, please provide an updated state machine drawing like @bpwilcox did in #1108. You can just upload a picture of a hand-drawn diagram if that's easier.